### ISSUE TITLE
HLSL: Function with return value but no return statement still parses to SPIR-V

### ISSUE THREAD

**Issue by redorav** [KhronosGroup/glslang#491]
Imagine a function like the following:

```
float MyFunc()
{
    // Code here without a return statement
}
```

This does not issue a warning like it does in glsl where it complains about the function not returning a value:

ERROR: example.glsl:32: '' : function does not return a value: MyFunc()

Addendum: this only happens if the function is not used.

**Comment by johnkslang** [comment]
This one is probably worth checking for.

However, note that glslang is not currently aiming for being an HLSL validator, so would not like to see a large set of issues in that space. It is worth checking for the ones that lead to problems in the stack though.

**Comment by johnkslang** [comment]
This is fixed by a305166ea427ba0618cde54ea03e647d73ab0044.
